config/packages/security.yaml

```
@@ -1,30 +1,43 @@
 security:
-    encoders:
-        App\Entity\Contractor:
-            algorithm: auto
+  encoders:
+    App\Entity\Contractor:
+      algorithm: auto
+    App\Entity\User:
+      algorithm: auto
 
-    # https://symfony.com/doc/current/security.html#where-do-users-come-from-user-providers
-    providers:
-        # used to reload user from session & other features (e.g. switch_user)
-        app_user_provider:
-            entity:
-                class: App\Entity\Contractor
-                property: username
-        # used to reload user from session & other features (e.g. switch_user)
-    firewalls:
-        dev:
-            pattern: ^/(_(profiler|wdt)|css|images|js)/
-            security: false
-        main:
-            anonymous: true
-            guard:
-                authenticators:
-                    - App\Security\ContractorAuthenticator
-            logout:
-                path: app_logout
-                invalidate_session: false
-                success_handler: App\Security\LogoutHandler
+  providers:
+    app_user_provider:
+      entity: { class: App\Entity\Contractor, property: username }
+    app_admin_provider:
+      entity: { class: App\Entity\User, property: email }
+  firewalls:
+    dev:
+      pattern: ^/(_(profiler|wdt)|css|images|js)/
+      security: false
+    admin:
+        pattern: ^/admin/
+        anonymous: true
+        provider: app_admin_provider
+        guard:
+          authenticators:
+            - App\Security\UserAuthenticator
+        logout:
+          path: admin_logout
+          invalidate_session: false
+          success_handler: App\Security\LogoutHandler
+    main:
+      anonymous: true
+      provider: app_user_provider
+      guard:
+        authenticators:
+          - App\Security\ContractorAuthenticator
+      logout:
+        path: app_logout
+        invalidate_session: false
+        success_handler: App\Security\LogoutHandler
 
-    access_control:
-          - { path: ^/contractor, roles: ROLE_CONTRACTOR }
-          - { path: ^/api/contractor, roles: ROLE_CONTRACTOR }
+  access_control:
+    - { path: /admin/login, role: IS_AUTHENTICATED_ANONYMOUSLY}
```

```suggestion
    - { path: ^/admin/login, role: IS_AUTHENTICATED_ANONYMOUSLY}
```
It works like regular expressions. ^ means beginning of the route. 
If you leve it /admin/login it may work on other longer routes like /new-route/admin/login. So it is better to specify it like this.